src/map.rs

```diff
@@ -168,6 +160,39 @@ impl<'a, T> PutResult<'a, T> {
     }
 }
 
+/// The error type for the [`HashMap::try_insert`] method.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct TryInsertError<'a, V> {
+    /// A reference to the old value mapped to the key.
```

Comment:
I think I would rename this to `current`, since it is not actually old (as we didn't replace it).